drivers/soundwire/slave.c

```diff
@@ -39,12 +39,12 @@ int sdw_slave_add(struct sdw_bus *bus,
 
 	if (id->unique_id == SDW_IGNORED_UNIQUE_ID) {
 		/* name shall be sdw:link:mfg:part:class */
-		dev_set_name(&slave->dev, "sdw:%x:%x:%x:%x",
+		dev_set_name(&slave->dev, "sdw:%x:%04x:%04x:%02x",
 			     bus->link_id, id->mfg_id, id->part_id,
 			     id->class_id);
 	} else {
 		/* name shall be sdw:link:mfg:part:class:unique */
-		dev_set_name(&slave->dev, "sdw:%x:%x:%x:%x:%x",
+		dev_set_name(&slave->dev, "sdw:%x:%04x:%04x:%02x:%01x",
```

Comment:
Is `%01x` necessary?

Reply:
It seems this is functionally the same as "%x", but for some reason "%01x" is pretty widely used. So I'm guessing these is more for code documentation purposes.

Reply:
I don't think it makes a difference, this was more to show in code review that the bitfield contains 4 bits.